### PR TITLE
feat(PL-6287): extract config to catalog

### PR DESCRIPTION
**Pull request by silphid**
This PR allows `joy` cli (and eventually `joy-generator`) to read catalog chart refs from either `joy.yaml` or `catalog.yaml`. The goal is to eventually only read chart refs from `catalog.yaml` and remove those from `joy.yaml`.

NOTE: That other [catalog PR](https://github.com/nestoca/catalog/pull/53575) must be merged first.

---

> [!NOTE]
> **Medium Risk**
> Chart resolution affects every release render/deploy; wrong merge or missing `catalog.yaml` could change which charts resolve until catalog PR lands, though legacy `joy.yaml` remains the fallback.
> 
> **Overview**
> Introduces a **`Catalog` v1alpha1 API** (`catalog.yaml`) so chart **refs**, **default chart**, and optional **repo/revision** live in a catalog resource instead of only in `joy.yaml`.
> 
> **Config loading** now reads `catalog.yaml`, merges `spec.charts` into the in-memory catalog config via `MergeCatalogResourceIntoCatalogConfig` (resource **refs** replace the whole charts map when present; **default** overrides when set), then keeps validating merged charts. Stderr hints whether legacy `joy.yaml` or catalog resource values were used.
> 
> **Scheme** centralizes group/version/kind constants and adds **Catalog** GVK/GK; per-file kind constants were removed in favor of `scheme.go`.
> 
> **Helm `Chart` / `ChartFS`** move to `internal/helm`; `pkg/helm` type-aliases them so cache behavior stays the same.
> 
> Tests cover merge edge cases and `Load` with joy-only, catalog-only, combined priority, and invalid YAML.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ce9526c45c13c59ae88327778f2222201063cc9e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

- **New Features**
  - Added support for a `catalog.yaml` resource to define chart references and a default chart.
  - Catalog chart mappings can now be merged across configuration sources, with resource-defined values taking priority when overlapping.
  - Chart repository URLs now more reliably resolve to OCI-style URLs, including when the scheme is omitted.

- **Bug Fixes**
  - Improved loading and validation behavior for both the catalog and base catalog configuration, with clearer error handling for invalid inputs.

- **Tests**
  - Added coverage for merging, priority behavior, empty/partial configs, and invalid configuration cases.

### CODE REVIEW

File: api/v1alpha1/scheme.go [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
 
 const (
-	KindRelease     = "Release"
-	KindEnvironment = "Environment"
-	KindProject     = "Project"
+	Group           = "joy.nesto.ca"
+	Version         = "v1alpha1"
+	ReleaseKind     = "Release"
```
Comment (davidmdm) [inline api/v1alpha1/scheme.go:10]: noooooooooooooooooooooooooo

when accessing vars, you want them to be searchable, aka what kinds are available. It is more searchable to prefix than to suffix.

File: internal/config/config.go [hunk 2]
```diff
@@ -190,19 +192,41 @@ func Load(ctx context.Context, configDir, catalogDir string) (*Config, error) {
...
+	}
+
 	if err := cfg.Validate(); err != nil {
 		return nil, fmt.Errorf("validating config: %w", err)
 	}
 
 	return &cfg, nil
 }
 
+func MergeCatalogResourceIntoCatalogConfig(catalogConfig *Catalog, catalogResource *v1alpha1.Catalog) error {
```
Comment (davidmdm) [inline internal/config/config.go:217]: I think the implementation should be simple precedence.

```go
if refs := catalogResource.Spec.Charts.Refs; refs != nil {
  catalogConfig.Charts = refs
}
if defaultRef := catalogResoure.Spec.Charts.Default; defaultRef != "" {
  catalogConfig.DefaultChartRef = defaultRef
}
```

File: internal/config/config_test.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+	cases := []struct {
+		name     string
+		joyYAML  string
+		catalogYAML string
+		expected Catalog
+	}{
+		{
+			name: "joy.yaml only",
+			joyYAML: `
```
Comment (davidmdm) [inline internal/config/config_test.go]: Purely aesthetic nitpick, when writing yaml, instead of writing pure yaml that totally breaks the Go indentation, you can write a halfway between json and yaml, that allows you to add indentation...

```go
		{
			name: "joy.yaml only",
			joyYAML: `{
				charts: {
					joy-chart: {
						repoUrl: "joy.example.com",
						name: "charts/joy",
						...
					}
				}`
			}
```

**Review by davidmdm — APPROVED** [review]
Pre-approving but check out comments for fun!

**Review by j-martin — APPROVED** [review]
Let's not forget to upgrade the code that uses this:

https://github.com/search?q=org%3Anestoca+%22nestoca%2Fjoy%22+language%3AGo+NOT+repo%3Anestoca%2Fjoy&type=code&ref=advsearch&l=Go
